### ISSUE TITLE
JPEG XL Support

### ISSUE THREAD

**Issue by SalOne22** [SalOne22/rimage#17]
Add support for JPEG XL format.

## Description:
We need to add support for the JPEG XL image format to our tool to allow users to convert their images to this format. This will provide more flexibility and options for users who want to use JPEG XL images in their projects.

## Required Library:
- jpegxl-rs (https://crates.io/crates/jpegxl-rs)

## Steps to Implement:
1. Install the `jpegxl-rs` crate using Cargo.
2. Implement support for the JPEG XL format in our code.
3. Add a new option to our CLI tool that allows users to convert images to the JPEG XL format.
4. Test the new functionality to ensure it works correctly.

## Expected Behavior:
Users should be able to convert their images to the JPEG XL format using our tool, providing more flexibility and options for their projects.

## Actual Behavior:
Currently, our tool does not support the JPEG XL format, which limits its usefulness for users who need to use this format.

## Note:
JPEG XL is a new image format that promises to provide better compression and quality than existing formats like JPEG and PNG. Adding support for it will make our tool more valuable to users.

**Comment by roopeshsn** [comment]
Hi, @SalOne22! I would like to work on this issue.

**Comment by SalOne22** [comment]
Ok, no problem, create your fork and work on it, thanks for contributing to my project!

**Comment by SalOne22** [comment]
Little note for develop, all image data is u8 vector in rgba format, this isn't mentioned in docs and I will fix it soon

**Comment by SalOne22** [comment]
@roopeshsn Hi, here is some updates in Decoder struct and ImageData, `Decoder` now accept opened file, and `ImageData` now stores bytes as `Box<[u8]>`

**Comment by roopeshsn** [comment]
Noted, @SalOne22. Possibly I'll complete it before this weekend.

**Comment by roopeshsn** [comment]
Hi, @SalOne22! After I had written decode and encode functions for .jxl format I used `cargo build`. I got this error message,
```
error: failed to run custom build command for `jpegxl-sys v0.8.1+libjxl-0.8.1`

Caused by:
  process didn't exit successfully: `/home/pando-roopesh/ross/rimage/target/debug/build/jpegxl-sys-e26c0981d48cdce1/build-script-build` (exit status: 101)
  --- stdout
  cargo:rerun-if-env-changed=LIBJXL_NO_PKG_CONFIG
  cargo:rerun-if-env-changed=PKG_CONFIG_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG
  cargo:rerun-if-env-changed=PKG_CONFIG
  cargo:rerun-if-env-changed=LIBJXL_STATIC
  cargo:rerun-if-env-changed=LIBJXL_DYNAMIC
  cargo:rerun-if-env-changed=PKG_CONFIG_ALL_STATIC
  cargo:rerun-if-env-changed=PKG_CONFIG_ALL_DYNAMIC
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_PATH
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_LIBDIR
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_SYSROOT_DIR
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR

  --- stderr
  thread 'main' panicked at 'Cannot find `libjxl` with version >= 0.8.1', /home/pando-roopesh/.cargo/registry/src/github.com-1ecc6299db9ec823/jpegxl-sys-0.8.1+libjxl-0.8.1/build.rs:20:37
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
```

It looks like the `jpegxl-rs` crate depends on `libjxl` library. Should I need to manually install `libjxl`?

**Comment by SalOne22** [comment]
Hi, @roopeshsn, i checked the docs for jpegxl, you can enable `vendored` feature to build the libjxl and statically link it, or if this doesn't work install libjxl manually.

**Comment by SalOne22** [comment]
Hey @roopeshsn, how is the development of jpegxl support coming along? During this time, I managed to completely rewrite the API, so most likely you will need to rewrite the code a bit. If you don't have enough time for that then I can use your code and tweak it for the new API, do you support that?

**Comment by SalOne22** [comment]
Implemented in version 0.9.0
